### PR TITLE
GKE also requires SAR endpoints

### LINKED ISSUE #76726: SubjectAccessReview endpoint should exist for providers other than GKE explicitly
**What would you like to be added**:
The function `WaitForNamedAuthorizationUpdate` in `test/e2e/framework/auth/helpers.go` should check for GKE and allow a missing SubjectAccessReview endpoint. If we get a 404 on another provider an error should still be returned.

GKE doesn't enable the SAR endpoint.  Without this endpoint, we cannot determine if the policy engine has adjusted as expected.  In this case, we currently wait one second and hope it's up to date without returning an error and that it has adjusted as expected.

**Why is this needed**:
A 404 on that endpoint is not expected behavior for any provider other than GKE so it doesn't make any sense to implicitly allow 404s and *hope* that everything is ok. If it is a special, allowable, condition for GKE we should make that explicit in the code.

The problem with doing it now is that checking the provider requires access to the TestContext which forces us to import the entire `test/e2e/framework` package. This goes against the goal of #76206 

At the same time, we don't want to cause every test/caller of this to have to (1) know that GKE acts differently in this situation and code that one special path. This may be the lesser of the evils though, but regardless of how it is accomplished, at some point, the code should be validating this flow.

Open to other solutions to this; maybe a specific test to test this code path (in which case this helper can be assumed to not be responsible for it?) Not sure if that is considered sufficient as it creates an awkward coupling between a helper function's assumptions and the existence of particular test.

### PR DESCRIPTION
**Pull request by johnSchnake**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
There was a specific error flow that was commented as only applying
to GKE. This was never tested specifically for GKE (only commented
as such) but that seems to be out of date and can be removed. If
the SAR endpoint does not exist it should be considered an error.

**Which issue(s) this PR fixes**:
Fixes #76726

**Special notes for your reviewer**:
See https://cloud.google.com/kubernetes-engine/docs/reference/api-permissions and just search for `subjectaccessreview`. You'll see there are subjectaccessreview, selfsubjectaccessreview, and localsubjectaccessreview endpoints.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/sig testing
/area test
/area test-framework

### CODE REVIEW

**Review by neolit123 — COMMENTED** [review]
SGTM, thanks.

/priority backlog
/approve
/sig gcp

**Review by timothysc — APPROVED** [review]
/lgtm 
/approve 
/hold 

@kubernetes/sig-gcp-pr-reviews  
Timeout 1 week.

**Comment by timothysc** [comment]
@johnSchnake you can unhold PRs that are pending on timeout if approvers are ok with it. 
/hold cancel

**Comment by johnSchnake** [comment]
@timothysc Oh OK, thanks. I was assuming the person who placed the hold would do so. Thanks for the clarification on convention; still new to the reviewer game :)

**Pull request merged**
